### PR TITLE
Database documentation

### PR DESCRIPTION
**Pull request by BorisKest**
Added basic documentation for the database and Drift library.

### CODE REVIEW

File: src/content/docs/essentials/database.mdx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+title: Database
+description: Database and Drift
+---
+
+In almost any project we deal with data, often in substantial quantities. In such cases, it is important
+to decide on the tools that will be used. We need that the data is reliably stored and that we can access
+it without dancing around the fire with tambourines. When working with a cross-platform, it is also important
+to consider that the user can access data on different devices. And in all of this, the lightweight fighter
+[SQLite](#why-sqlite) will come to our aid, who will be "guide" on the right path by
+[Drift library](#what-is-drift).
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: Our projects mainly focus on interacting with the backend through message exchange. At times, we need to store data for caching or similar reasons. While there's a wide range of options available, various factors must be considered carefully. After evaluating numerous libraries, we've selected [SQLite](#sqlite) as our preferred choice.

File: src/content/docs/essentials/database.mdx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+## Why SQLite?
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: ## SQLite

File: src/content/docs/essentials/database.mdx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+SQLite is selected for data processing due:
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: SQLite is a compact, relational database management system designed for efficient disk-based operations. It's widely utilized in a variety of settings, including desktop and mobile applications, IoT devices, and web environments. With its deployment in millions of production applications globally, SQLite's robustness is further reinforced by an extensive suite of over one billion test cases, ensuring a high level of reliability and minimal error incidence.

File: src/content/docs/essentials/database.mdx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  devices, which is important when working with a multi-platform framework such as
+  Flutter because it allows seamless integration between different platforms.
+
+- **Data Analysis:** Efficient for analyzing large datasets using SQL commands, making it a choice for
+  bioinformatics research.
+
+- **Cache for Enterprise Data:** Acts as a cache for enterprise content, reducing latency and network load.
+
+- **Data Transfer Format:** Serves as a compact, cross-platform container for transferring content between
+  systems.
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: Not needed

File: src/content/docs/essentials/database.mdx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+More about SQLite can be found [here](https://www.sqlite.org/whentouse.html).
+
+## What is Drift?
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: ## Drift

File: src/content/docs/essentials/database.mdx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+of the column.
+
+```dart
+class Spendings extends Table {
+    IntColumn get id => integer().autoIncrement()();
+    RealColumn get price => real().withDefault(const Constant(0.0))();
+    TextColumn get title => text()();
+    TextColumn get currency => text().nullable()();
+    DateTimeColumn get date => dateTime()();
+    BlobColumn get receiptImage => blob().nullable()();
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: It is considered a bad practise to store images in databases as BLOB. It is preferred to store links in fs.

File: src/content/docs/essentials/database.mdx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+}
+```
+
+detailed description of defining tables can be found
+[here](https://drift.simonbinder.eu/docs/getting-started/advanced_dart_tables/)
+
+### Data source and repositorie
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: This is not needed

File: src/content/docs/essentials/database.mdx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+or limit the data.
+
+```dart
+    final query = await database.select(spendings)
+        ..limit(10);
+```
+
+No one will stop us from combining everything together, but that's covered
+[here](https://drift.simonbinder.eu/docs/dart-api/select/).
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: To my mind, there is no need to tell about all the features of Drift. Leave only inserts, updates and selects (but make select with various modificators such as limit and ordering)

File: src/content/docs/essentials/database.mdx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+## Executors
+
+In Drift, an executor is a component responsible for handling and executing database queries.
+Drift employs different executors based on the platform to ensure optimal performance and compatibility.
+These platform-specific executors are designed to handle database interactions in a way that aligns
+with the nuances and requirements of each particular platform, providing efficient and seamless
+functionality across various devices.
+
+#### Whats the deal with package?
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: ### Local package

File: src/content/docs/essentials/database.mdx [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+        ..limit(10);
+```
+
+No one will stop us from combining everything together, but that's covered
+[here](https://drift.simonbinder.eu/docs/dart-api/select/).
+
+### Migrations
+
+It's often challenging to foresee precisely what data should be in the table right away. In the
+future, new fields or tables may emerge, and for such changes, a migration tool exists.
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: Predicting initial data requirements for a table can be complex, as data needs often evolve, requiring new fields or tables. Migration tools are crucial for adapting the database structure to these changing requirements, ensuring flexibility and robustness.

File: src/content/docs/essentials/database.mdx [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+
+[Migration API](https://drift.simonbinder.eu/docs/migrations/)
+
+## Executors
+
+In Drift, an executor is a component responsible for handling and executing database queries.
+Drift employs different executors based on the platform to ensure optimal performance and compatibility.
+These platform-specific executors are designed to handle database interactions in a way that aligns
+with the nuances and requirements of each particular platform, providing efficient and seamless
+functionality across various devices.
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: Remove

File: src/content/docs/essentials/database.mdx [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+
+#### Whats the deal with package?
+
+When moving the database into a separate package with Drift and Flutter, it also simplifies working with
+code generation. The generated code, which creates classes for interacting with the database based on your
+Drift files, becomes part of this standalone package. This contributes to better project organization,
+reduces dependencies, and enhances maintainability.
```
Comment (hawkkiller) [inline src/content/docs/essentials/database.mdx]: The database schema is moved to a different package called `database`. There are a few reasons for this: the schema itself doesn't change frequently, so there is no need to spend time during the codegen. Moreover, it improves project organization as everything related to database is grouped in different package. This package also provides executors logic.

**Executor** is a component tasked with managing and executing database queries. To guarantee optimal performance and compatibility, Drift utilizes different executors tailored to each platform. These platform-specific executors are engineered to align with the unique nuances and requirements of their respective platforms, ensuring efficient and seamless database interactions across a range of devices.

**Review by hawkkiller — CHANGES_REQUESTED** [review]

**Review by BorisKest — COMMENTED** [review]
Added requested changes.

**Pull request merged**